airflow/utils/timezone.py

```diff
@@ -89,10 +89,11 @@ def convert_to_utc(value):
     if not value:
         return value
 
-    if not is_localized(value):
-        value = pendulum.instance(value, TIMEZONE)
-
-    return value.astimezone(utc)
+    return (
```

Comment:
The issue causing failing `test_following_previous_schedule` with this change.  
The problem is that `pendulum.instance(start)` does not convert correctly.

This seems to work:
```suggestion
    if isinstance(value, pendulum.DateTime):
        return value.astimezone(utc)
    elif not is_localized(value):
        return pendulum.instance(value, TIMEZONE)
    else:
        return utc.convert(value)
```

Reply:
OK my suggested change to `convert_to_utc` only fixed issue locally on my py3.6 env.  when i pushed changes on my fork, the tests still failed ....very weird ... i suspect possibly it could be the py3.5 issues referenced here: https://pendulum.eustace.io/docs/#timezones.

i wonder if this change should be combined or replaced with a change to use pendulum everywhere, instead of the piecemeal status quo.  if datetimes were everywhere always pendulum, it would be a lot easier to manage this kind of upgrade, and easier to write tests.

Reply:
OK so i finally installed py3.5 locally to see what’s going on.

`Timezone.convert()` does not work right in pendulum 2.0 on python 3.5.  It does not respect `dst_rule`.  So the problem is with the structure of the test `test_following_previous_schedule` itself.

In `test_following_previous_schedule`, `start` is defined like so:  ```python
start = local_tz.convert(datetime.datetime(2018, 10, 28, 2, 55), dst_rule=pendulum.PRE_TRANSITION)```

On py3.5 with pendulum 2.0, `start` gets the wrong utcoffset: 1 hour;  in all other combinations (i.e. 3.5/1.5, 3.6/1.5, 3.7/1.5, 3.6/2.0, 3.7/2.0), the offset is 2 hours (which is the correct offset).

So, I think what we need to do is instead define `start` like so:
```
start = pendulum.datetime(2018, 10, 28, 2, 55, dst_rule=pendulum.PRE_TRANSITION, tz=local_tz)```

And additionally, we need to review the code, to see where we might be using `Timezone.convert`, and make changes as appropriate.

Additional note:
On py3.5 with pendulum 2.0 you can see that dst_rule is not respected in `convert`:
```python
import pendulum, datetime

local_tz = pendulum.timezone('Europe/Zurich')
UTC = pendulum.timezone('UTC')
start_post = local_tz.convert(datetime.datetime(2018, 10, 28, 2, 55), dst_rule=pendulum.POST_TRANSITION)
start_pre = local_tz.convert(datetime.datetime(2018, 10, 28, 2, 55), dst_rule=pendulum.PRE_TRANSITION)
print(start_post.isoformat())
print(start_pre.isoformat())
```